### PR TITLE
(FM-2752) Add/update travis with modulesync

### PR DESCRIPTION
**Pull request by cyberious**

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+clone_depth: 10
+init:
+- SET
+- 'mkdir C:\ProgramData\PuppetLabs\code && exit 0'
+- 'mkdir C:\ProgramData\PuppetLabs\facter && exit 0'
+- 'mkdir C:\ProgramData\PuppetLabs\hiera && exit 0'
+- 'mkdir C:\ProgramData\PuppetLabs\puppet\var && exit 0'
+environment:
+  matrix:
+  - PUPPET_GEM_VERSION: 3.7.5
```
Comment (ferventcoder) [inline appveyor.yml]: All the way back to 3.3.0 you need https://github.com/puppetlabs/puppetlabs-registry/blob/master/metadata.json#L32. I'd do that, 3.8.1 and 4.1.0.

Comment (cyberious) [inline appveyor.yml]: In the long term perhaps, but this is an initial spike for current development, I would like to get our tests working in ~>3.7 and ~>4.0 first than we can take on the load to get them to work back to 3.3

File: .travis.yml [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
+script: "bundle exec rake validate && bundle exec rake lint && bundle exec rake spec SPEC_OPTS='--color --format documentation'"
 matrix:
   fast_finish: true
+  include:
+  - rvm: 1.9.3
+    env: PUPPET_GEM_VERSION="~> 3.0"
+  - rvm: 2.1.5
+    env: PUPPET_GEM_VERSION="~> 3.0"
+  - rvm: 2.1.5
+    env: PUPPET_GEM_VERSIONS="~> 4.0"
```
Comment (Iristyle) [inline .travis.yml]: @cyberious we have a straggler `s`

Comment (ferventcoder) [inline .travis.yml]: Ping @cyberious

**Comment by cyberious** [comment]
@ferventcoder @Iristyle updated

**Comment by ferventcoder** [comment]
As long as we are not losing sight of the fact that this needs to port back to testing against 3.3.0, I'm +1

**Pull request merged**